### ISSUE TITLE
Shebang line should not be passed to rules as a comment

### ISSUE THREAD

**Issue by gcochard** [eslint/eslint#1352]
Currently, the shebang is being reported as a `LineComment` to the rule engine, with the node value having dropped the `#` so that it appears as `!/usr/bin/env node`.

It would be preferable to either drop it from the program altogether, make a new node type such as `Shebang`, or pass the empty string as the node value.

**Comment by Dzenly** [comment]
This issue breaks my code:
```
#!/usr/bin/env bash
':'; // # comment; exec /usr/bin/env node "$0" "$@"
```
If there is newline after shebang, bash don't start the node.

**Comment by platinumazure** [comment]
@Dzenly Please open a new issue and provide repro steps. All ESLint does here is convert the shebang into a comment *for parsing*; it should not edit the contents of your file in general, unless a rule is doing some sort of autofix on that line (which is pretty unlikely).

**Comment by Dzenly** [comment]
I am about --fix option. My webstorm is setup for eslint --fix on file save and now I can not use webstorm for this file.

**Comment by platinumazure** [comment]
@Dzenly What you are describing is likely not related to this issue. Please open a new issue and please fill out [this template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md) (hopefully will be pre-rendered into the new issue form). There are a lot of pieces of information that we need, which you aren't providing here. Again, this should be further discussed in a new issue, not this issue. Thanks!
